cmd/main.go

```diff
@@ -36,7 +45,41 @@ var (
 )
 
 var saramaLogger = log.New(io.Discard, "[Sarama] ", log.LstdFlags)
+
+func InitTracerProvider(ActivateTracing bool, JaegerExporterTracing bool, OtlpExporterTracing bool) *sdktrace.TracerProvider {
+	resources, _ := resource.New(context.Background(),
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithProcess(), // This option configures a set of Detectors that discover process information
+	)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}))
+	if ActivateTracing {
+		exporter := exporterTracing(JaegerExporterTracing, OtlpExporterTracing)
+		tp := sdktrace.NewTracerProvider(
+			sdktrace.WithResource(resources),
+			sdktrace.WithBatcher(exporter),
+		)
+		otel.SetTracerProvider(tp)
+		return tp
+
+	}
+	otel.SetTracerProvider(trace.NewNoopTracerProvider())
+	return nil
+}
+
+func exporterTracing(JaegerExporterTracing bool, OtlpExporterTracing bool) sdktrace.SpanExporter {
+	//Could not use OTEL_TRACES_EXPORTER https://github.com/open-telemetry/opentelemetry-go/issues/2310
+	var exporter sdktrace.SpanExporter
+	//If the env variables needed are defined we set the exporter to Jaeger else it is an opentelemetry exporter
+	if JaegerExporterTracing {
+		exporter, _ = jaeger.New(jaeger.WithAgentEndpoint()) //from env variable https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/jaeger#environment-variables
+	} else if OtlpExporterTracing {
+		exporter, _ = otlptracegrpc.New(context.Background(), otlptracegrpc.WithInsecure()) //from env variable https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/README.md
```

Comment:
We should be secure by default, so I question the use `otlptracegrpc.WithInsecure()`.  Perhaps an environment variable should give the user the ability to configure this, if they wish.

Reply:
by removing otlptracegrpc.WithInsecure() if we put the right env variable this is secure